sympy/functions/elementary/tests/test_exponential.py

```diff
@@ -648,3 +649,29 @@ def test_issue_9116():
 
     assert ln(n).is_nonnegative is True
     assert log(n).is_nonnegative is True
+
+
+def test_lambertw_identities():
+    def entry(i, j, simp=False, k=0):
+        x = (i - 5) / 2
+        y = (j - 5) / 2
+        z = (x + y*I)
+        ret = LambertW(z*exp(z), k, evaluate=False)
+        if not simp:
+            return ret
+        return ret._extract_identity()
+
+    def numerically_close(m1, m2):
+        return all(abs((a-b).evalf()) < 10**-15 for a, b in zip(m1, m2))
+
+    m1 = Matrix(11, 11, lambda i, j: entry(i, j, simp=False, k=0))
+    m2 = Matrix(11, 11, lambda i, j: entry(i, j, simp=True, k=0))
+    assert numerically_close(m1, m2)
```

Comment:
This is a strangely organised test. Are the matrices relevant somehow? I would just write this as nested loops:
```python
for k in range(3):
    for x in range(11):
        for y in range(11):
            z = x+I*y
            ...
            assert (w - w._extract_identity()) < 1e-15
```

Also why is the test only for approximate?

Reply:
There are too many points to test, and the easiest way is comparing with the numeric results such that the simplification does not give some artificially wrong results.